cmd/oras/internal/display/progress/manager.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package progress
+
+import (
+	"errors"
+	"os"
+	"sync"
+	"time"
+
+	"oras.land/oras/cmd/oras/internal/display/console"
+)
+
+// BufferSize is the size of the status channel buffer.
+const BufferSize = 20
+
+var errManagerStopped = errors.New("progress output manage has already been stopped")
+
+// Status is print message channel
+type Status chan<- *status
+
+// Manager is progress view master
+type Manager interface {
+	Add() (Status, error)
+	Close() error
+}
+
+const bufFlushDuration = 200 * time.Millisecond
+
+type manager struct {
+	status       []*status
+	statusLock   sync.RWMutex
+	console      *console.Console
+	updating     sync.WaitGroup
+	renderDone   chan struct{}
+	renderClosed chan struct{}
+}
+
+// NewManager initialized a new progress manager.
+func NewManager(f *os.File) (Manager, error) {
+	c, err := console.New(f)
+	if err != nil {
+		return nil, err
+	}
+	m := &manager{
+		console:      c,
+		renderDone:   make(chan struct{}),
+		renderClosed: make(chan struct{}),
+	}
+	m.start()
+	return m, nil
+}
+
+func (m *manager) start() {
+	m.console.Save()
+	renderTicker := time.NewTicker(bufFlushDuration)
+	go func() {
+		defer m.console.Restore()
+		defer renderTicker.Stop()
+		for {
+			select {
+			case <-m.renderDone:
+				m.updating.Wait()
```

Comment:
`m.updating.Wait()` should be between line 138 and 139 before `close(m.renderDone)` so that the `manager` can keep rendering while waiting for all status to complete.

Reply:
Nice catch. Will revert.